### PR TITLE
tests: move test cases of privilege package to integration test (part2)

### LINKED ISSUE #47149: Move IT in `privilege/` to explaintest, files number 4

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #47149 

Problem Summary:

### What is changed and how it works?

Move more unit tests to integration tests.

### Check List

Tests 

- [ ] Unit test
- [X] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/integrationtest/t/privilege/privileges.test [hunk 1]
```diff
@@ -491,4 +494,314 @@ GRANT CREATE TEMPORARY TABLES,DELETE,EXECUTE,INSERT,SELECT,SHOW VIEW,TRIGGER,UPD
...
+SHOW GRANTS FOR 'tsg_u2';
+SET ROLE tsg_r1;
+SHOW GRANTS;
+SHOW GRANTS FOR CURRENT_USER();
+SHOW GRANTS FOR 'tsg_u2';
+disconnect tsg_u2;
+connection default;
+
+
+# TestGetEncodedPassword
```
Comment (bb7133) [inline tests/integrationtest/t/privilege/privileges.test:775]: Seems we cannot port this case directly.

File: privilege/privileges/privileges_test.go [hunk 2]
```diff
@@ -773,31 +691,6 @@ func TestUseDB(t *testing.T) {
 	require.Error(t, err)
 }
 
-// func TestRevokePrivileges(t *testing.T) {
```
Comment (bb7133) [inline privilege/privileges/privileges_test.go:776]: We can just keep it and try to un-comment this case in the future.

File: tests/integrationtest/t/privilege/privileges.test (hunk 1 above)
Comment (tiancaiamao) [reply tests/integrationtest/t/privilege/privileges.test:775]: Why not?

https://github.com/pingcap/tidb/pull/47292/files#diff-8b507d8e9d7f506b565cbedfd8e7c860aa76099537d36ed9cdeaf931d71d64c0L1171

https://github.com/pingcap/tidb/pull/47292/files#diff-d305408c493422010fcbfbf7a67ed9e3ccdd0161d3eb5c74797535e13ab50a53R548-R550

File: privilege/privileges/privileges_test.go (hunk 2 above)
Comment (tiancaiamao) [reply privilege/privileges/privileges_test.go:776]: https://github.com/pingcap/tidb/blob/6db372a10b48b4640a4722c0c91e02477984dc68/tests/integrationtest/t/privilege/privileges.test#L231C1-L256

It's already moved and in the last PR I just comment it and forget to clean up... so this PR I clean it up here.

**Review by Defined2014 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**